crates/bevy_render/src/render_resource/pipeline.rs

```diff
@@ -158,6 +158,24 @@ impl VertexBufferLayout {
             attributes,
         }
     }
+
+    /// Returns a [`VertexBufferLayout`] with the shader location of every attribute offset by
+    /// `location`.
+    pub fn offset_locations(self, location: u32) -> Self {
+        Self {
+            array_stride: self.array_stride,
+            step_mode: self.step_mode,
+            attributes: self
+                .attributes
+                .into_iter()
+                .map(|attr| VertexAttribute {
+                    format: attr.format,
+                    offset: attr.offset,
+                    shader_location: attr.shader_location + location,
+                })
+                .collect(),
+        }
```

Comment:
Minor thought: What about mutating the list and then using `std::mem::take`. Not exactly sure how the generated code looks like. I'll check in a second.